### ISSUE TITLE
A bilinearly resampled mask lets detections through inside the zero region

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4102]
#4098 made `MultiResolutionDetector`'s `mask` argument work. The mask is resampled onto each
pyramid level with bilinear interpolation and multiplied into the response before non-maxima
suppression, mirroring `ScaleSpaceDetector`. The bilinear ramp at the mask edge means a feature
lying entirely inside the zero region can still be detected.

## Reproduction

A blob centred at `x ~ 29.5`, wholly inside the masked-out half, with a mask that allows only
`x >= 32`:

```python
import torch
from kornia.feature import MultiResolutionDetector
from kornia.feature.responses import BlobHessian

img = torch.zeros(1, 1, 64, 64)
img[:, :, 30:34, 28:32] = 1.0                     # blob centre x ~ 29.5

mask = torch.zeros(1, 1, 64, 64)
mask[:, :, :, 32:] = 1.0                          # allow only x >= 32

lafs, resps = MultiResolutionDetector(BlobHessian(), num_features=5)(img, mask)
print([round(v, 2) for v in lafs[0][resps[0] != 0][:, 0, 2].tolist()])
```

```
[30.97, 31.29, 31.29]
```

Three detections, all on the side the mask zeroes.

## Cause

`kornia/feature/scale_space_detector.py:81`

```python
return F.interpolate(mask, list(ref.shape[-2:]), mode="bilinear", align_corners=False).to(ref.dtype)
```

At a level downsampled by `sqrt(2)^k`, a level pixel straddling the mask boundary gets a weight
strictly between 0 and 1, so a strong response just inside the zero region survives the multiply
and the threshold. The coarser the level, the further in this reaches: one coarse-level pixel is
about 4 image pixels for the default five-level configuration.

The `detect` docstring already says the edge "softens by about one level pixel", and the measured
1.03 px is within that, so this is a documented approximation rather than a contradiction. It is
still worth deciding deliberately, because "a zero region suppresses detections" is what a caller
reading the argument name expects, and a segmentation or privacy mask is exactly the case where
leaking a detection matters.

## Options

1. Threshold the resampled mask, e.g. `(resized > 0).to(dtype)` for a strict interpretation, or
   `> 0.5` for a majority one. Cheap, and makes the zero region exact.
2. Resample with `mode="nearest"` (or `area`) for a boolean mask and keep bilinear for a weight
   mask, keyed on the input dtype.
3. Leave it and keep the docstring caveat.

`ScaleSpaceDetector` shares `_resize_mask`, so whichever is chosen applies to both. On `main` its
octave geometry happened not to leak on this input (the same blob is detected at exactly `x = 32`),
but the mechanism is the same.

Found while reviewing #4098; deliberately left out of it as out of scope.
